addons/project/tests/test_project_base.py

```diff
@@ -385,3 +385,30 @@ def test_create_task_in_batch_with_email_cc(self):
         self.assertTrue(user_b.partner_id in task_2.message_partner_ids)
         self.assertTrue(user_c.partner_id in task_2.message_partner_ids)
         self.assertFalse(partner in task_2.message_partner_ids)
+
+    def test_duplicate_project_with_tasks(self):
+        """ Test to check duplication of projects tasks active stats. """
+        self.env['project.task'].create({
+            "name": "Achived Task",
+            "active": False,
+            "project_id": self.project_pigs.id})
+
+        def _get_all_tasks(project_id):
+            return self.env['project.task'].with_context(active_test=False).search([("project_id", "=", project_id)])
+
+        # 1- Check if duplicated project have the same tasks, active state should be kept when duplicating an active project
+        duplicated_pigs_project = self.project_pigs.copy()
+        pigs_tasks = _get_all_tasks(self.project_pigs.id)
+        duplicated_pigs_tasks = _get_all_tasks(duplicated_pigs_project.id)
```

Comment:
But why make it dynamic?
Shouldn't we just create one project with one task and work with that?